provisioning/baremetal_pod.go

```diff
@@ -516,3 +517,32 @@ func ApplyNewOrUpdatedDeployment(client appsclientv1.DeploymentsGetter, targetNa
 
 	return nil
 }
+
+func getDeploymentCondition(deployment *appsv1.Deployment) appsv1.DeploymentConditionType {
+	for _, cond := range deployment.Status.Conditions {
+		if cond.Status == corev1.ConditionTrue {
+			return cond.Type
+		}
+	}
+	return appsv1.DeploymentProgressing
+}
+
+// Provide the current state of metal3 deployment
+func GetDeploymentState(client appsclientv1.DeploymentsGetter, targetNamespace string, config *metal3iov1alpha1.Provisioning) (appsv1.DeploymentConditionType, error) {
+	existing, err := client.Deployments(targetNamespace).Get(context.Background(), BaremetalDeploymentName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// Deployment is not available and if Provisioning CR is not available,
+		// then everything is fine
+		if config == nil {
+			return appsv1.DeploymentAvailable, nil
+		} else {
+			// Provisioning CR is present so the deployment rollout could be in early stages
+			return appsv1.DeploymentProgressing, err
```

Comment:
The error pretty much overrides the condition type that we are returning, so it would seem less misleading to just return the default value.

Reply:
Again, updated logic around this. Still, continuing to return the DeploymentCondition and error though.